core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -1064,6 +1064,12 @@ set_exception_visitor(void *node, void *user_data)
         bh_memcpy_s(curr_wasm_inst->cur_exception,
                     sizeof(curr_wasm_inst->cur_exception),
                     wasm_inst->cur_exception, sizeof(wasm_inst->cur_exception));
+
+#ifdef OS_ENABLE_BLOCK_INSN_INTERRUPT
+        bh_assert(curr_exec_env->handle);
+        os_thread_signal(curr_exec_env->handle, SIGUSR1);
```

Comment:
Should not use SIGUSR1 in non-platform related code, suggest to wrap a function `os_thread_killl`, and call that API instead.